packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -36,18 +36,42 @@ enum _ScaffoldSlot {
   statusBar,
 }
 
+/// Geometry information for scaffold components.
+///
+/// To get a [ValueNotifier] for the scaffold gemoetry call
+/// [Scaffold.geometryOf].
+@immutable
+class ScaffoldGeometry {
+  /// The rectangle in which the scaffold is laying out
+  /// [Scaffold.bottomNavigationBar].
+  /// Null when there is no bottom navigation bar..
+  final Rect bottomNavigationBar;
```

Comment:
This could probably just be a double giving the distance from some reference edge to the top of the navbar area. That would reduce the temptation to use this value to determine geometry that is best discovered by directly examining the navbar.

Reply:
done